### PR TITLE
Update support for Replication module 1.4.0

### PR DESCRIPTION
**Pull request by santhoshatdell**
# Description
Updated replication manifest files and sample array files to include v1.4.0 changes and new replication image.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/583 |

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

**Review by lukeatdell — DISMISSED** [review]

**Review by ChristianAtDell — APPROVED** [review]

File: pkg/modules/replication.go [hunk 1]
```diff
@@ -318,7 +320,8 @@ func getReplicaController(op utils.OperatorConfig, cr csmv1.ContainerStorageModu
 	replicaCount := "1"
 	retryMin := "1s"
 	retryMax := "5m"
-	replicaImage := "dellemc/dell-replication-controller:v1.3.1"
+	replicaImage := "dellemc/dell-replication-controller:v1.4.0"
+	replicaInitImage := "dellemc/dell-replication-init:v1.0.0"
```
Comment (rensyct) [inline pkg/modules/replication.go:324]: Why the image has to be hardcoded here?

**Review by rensyct — COMMENTED** [review]
Please ensure to change https://github.com/dell/csm-operator/blob/main/operatorconfig/moduleconfig/common/version-values.yaml#L13

**Comment by santhoshatdell** [comment]
> Please ensure to change https://github.com/dell/csm-operator/blob/main/operatorconfig/moduleconfig/common/version-values.yaml#L13

Updated version-values.yaml. Also updated this file with latest Observability module version as per [PR](https://github.com/dell/csm-operator/pull/173/files#diff-bf4be8b5216ab38d658455dc44f345e6a345a80a7a511f250133989f8d55f508).

File: pkg/modules/replication.go (hunk 1 above)
Comment (santhoshatdell) [reply pkg/modules/replication.go:324]: I do not know why these were hardcoded here. It should not in my opinion too.
I see that they are used as default vaules and then overwritten from user defined yaml. I see the same for Observability.go as well.

**Review by abhi16394 — APPROVED** [review]

**Review by ChristianAtDell — APPROVED** [review]

**Pull request merged**